terraform/modules/benchmark/service/run_benchmark.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+## Enable strict error handling
+set -eux
+
+# 1. Initialize s3cmd
+# 2. Run benchmark
+# 3. Upload benchmark results to s3 bucket
+
+## Configure s3cmd
+CFG=~/.s3cfg
+
+if [ ! -f "$CFG" ]; then
+    s3cmd --dump-config \
+        --host="$BENCHMARK_ENDPOINT" \
+        --host-bucket="%(bucket)s.$BENCHMARK_ENDPOINT" \
+        --access_key="$AWS_ACCESS_KEY_ID" \
+        --secret_key="$AWS_SECRET_ACCESS_KEY" \
+        --multipart-chunk-size-mb=4096 > "$CFG"
+    echo "Configuration file created at $CFG"
+else
+    echo "s3cmd Configuration file exist at $CFG"
+fi
+
+## Run actual benchmark
+ruby bench.rb --chain calibnet --tempdir ./tmp --daily
+ruby bench.rb --chain mainnet --tempdir ./tmp --daily
+
+## Upload benchmark result to s3 weekly file
+week_number=$(date +%W) # Week starting on Monday
+s3cmd get s3://"$BENCHMARK_BUCKET"/benchmark-results/weekly-results/weekly_result_"$week_number".csv /tmp/weekly_result_"$week_number".csv --force || 
+echo "Timestamp,Forest Version,Lotus Version,Chain,Metric,Forest Value,Lotus Value" > /tmp/weekly_result_"$week_number".csv
+tail -n +2 -q /chainsafe/result_*.csv >> /tmp/weekly_result_"$week_number".csv && s3cmd --acl-public put /tmp/weekly_result_"$week_number".csv s3://"$BENCHMARK_BUCKET"/benchmark-results/weekly-results/weekly_result_"$week_number".csv || exit 1
```

Comment:
What is the plan to deal with breaking changes? Looking at the existing file at https://forest-benchmarks.fra1.digitaloceanspaces.com/benchmark-results/all_results.csv, there were some at one point:
```
2023-07-05 23:43:49,0.11.0,1.23.4,calibnet,validate_online,11.8,8.0
2023-07-06 01:48:32,0.11.0,1.23.4,mainnet,import,1443,2094
2023-07-06 01:48:32,0.11.0,1.23.4,mainnet,validate_online,0.209,0.175
2023-07-06 02:36:54,0.11.0,1.23.4,calibnet,import,41.1,96
2023-07-06 02:36:54,0.11.0,1.23.4,calibnet,validate_online,2.5,2.6
2023-07-06 04:43:39,0.11.0,1.23.4,mainnet,import,1480,2082
2023-07-06 04:43:39,0.11.0,1.23.4,mainnet,validate_online,0.275,0.184
2023-07-06 16:07:48,0.11.1,1.23.4,calibnet,import_time,44.8 sec,102 sec
2023-07-06 16:07:48,0.11.1,1.23.4,calibnet,validation_time,11.7 tipsets/sec,7.4 tipsets/sec
2023-07-06 18:15:00,0.11.1,1.23.4,mainnet,import_time,1415 sec,2054 sec
2023-07-06 18:15:00,0.11.1,1.23.4,mainnet,validation_time,0.159 tipsets/sec,0.092 tipsets/sec
2023-07-06 16:07:48,0.11.1,1.23.4,calibnet,import_time,44.8 sec,102 sec
2023-07-06 16:07:48,0.11.1,1.23.4,calibnet,validation_time,11.7 tipsets/sec,7.4 tipsets/sec
2023-07-06 18:15:00,0.11.1,1.23.4,mainnet,import_time,1415 sec,2054 sec
2023-07-06 18:15:00,0.11.1,1.23.4,mainnet,validation_time,0.159 tipsets/sec,0.092 tipsets/sec
2023-07-06 19:03:26,0.11.1,1.23.4,calibnet,import_time,43.6 sec,101 sec
2023-07-06 19:03:26,0.11.1,1.23.4,calibnet,validation_time,12.3 tipsets/sec,7.6 tipsets/sec
2023-07-06 21:07:41,0.11.1,1.23.4,mainnet,import_time,1400 sec,2110 sec
2023-07-06 21:07:41,0.11.1,1.23.4,mainnet,validation_time,0.267 tipsets/sec,0.192 tipsets/sec
2023-07-06 16:07:48,0.11.1,1.23.4,calibnet,import_time,44.8 sec,102 sec
```
Should the former ones be cleared? What is the plan for the future?

Reply:
will take care in a separate PR